weave

```diff
@@ -512,6 +512,26 @@ container_ip() {
     esac
 }
 
+# Given a container name or short ID in $1, ensure the specified
+# container is running and then print its full ID to stdout. If
+# it's not running or doesn't exist, print an error to stderr and
+# return with an indicative non-zero exit code.
+full_container_id() {
+    if ! STATUS=$(docker inspect --format='{{.State.Running}} {{.Id}}' $1 2>/dev/null) ; then
+        echo "Error: No such container: $1" >&2
+        return 1
+    fi
+    case "$STATUS" in
+        true*)
+            echo "${STATUS#true }"
+            ;;
+        *)
+            echo "$1 container is not running." >&2
+            return 2
+            ;;
+    esac
+}
+
```

Comment:
I have [just](https://github.com/weaveworks/weave/commit/79eb4c298c004db7c637fd6a50216bd7224cfcf2) put this on master, but called it `container_id`, and changed `get_ipam_cidr` to only take one argument - the full container id. There was only one call site where we didn't have that, and now use the helper to find it.